helix-core/src/main/java/org/apache/helix/monitoring/metrics/AbnormalStatesMetricCollector.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.helix.monitoring.metrics;
+
+import javax.management.JMException;
+
+import org.apache.helix.HelixException;
+import org.apache.helix.monitoring.mbeans.MonitorDomainNames;
+import org.apache.helix.monitoring.metrics.implementation.RebalanceCounter;
+import org.apache.helix.monitoring.metrics.model.CountMetric;
+
+public class AbnormalStatesMetricCollector extends MetricCollector {
+  private static final String ABNORMAL_STATES_ENTITY_NAME = "AbnormalStates";
+
+  public enum AbnormalStatesMetricNames {
+    AbnormalStatePartitionCounter, RecoveryAttemptCounter
+  }
+
+  public AbnormalStatesMetricCollector(String clusterName, String stateModelDef) {
+    super(MonitorDomainNames.Rebalancer.name(), clusterName,
+        String.format("%s.%s", ABNORMAL_STATES_ENTITY_NAME, stateModelDef));
+    createMetrics();
+    if (clusterName != null) {
+      try {
+        register();
+      } catch (JMException e) {
+        throw new HelixException("Failed to register MBean for the AbnormalStatesMetricCollector",
```

Comment:
Hardcoded `AbnormalStatesMetricCollector` -> AbnormalStatesMetricCollector.class.getSimpleName()? Benefit is  when refactoring the class name we wouldn't miss this.